### ISSUE TITLE
[Feature]: Native application

### ISSUE THREAD

**Issue by mrjvs** [movie-web/movie-web#703]
### What feature do you want to add?

Native application. It needs be run on android and IOS. Other platforms like desktop would be cool too.
Needs to be made on top of react native. Since we need a JS runtime to actually run the scrapers.

### Why do you want to have this feature?

native applications allow for more sources, and a better experience on to the user.

**Comment by maeseoki** [comment]
Why on top of React Native? Won't there be a better approach to build it with Capacitor?

**Comment by mrjvs** [comment]
Capacitor is just a webview, it does not actually make a native app.

**Comment by maeseoki** [comment]
> Capacitor is just a webview, it does not actually make a native app.

I know, with native plugins, but since the whole app is under React dom, seems a simpler task to run it under capacitor. Performance will be pretty the same.

**Comment by mrjvs** [comment]
@maeseoki 
If we copy paste it into a webview, there would be no benefits over doing it on a native app.
benefits of a native app are:
 - actual native components, speedy
 - no need to make compatible with mouse and keyboard, so its optimized for gestures
 - no security related to browsers, so can do anything with video player
 - no limitations for video formats, we can play anything

**Comment by timothymiller** [comment]
I can help make the native app

**Comment by maeseoki** [comment]
@mrjvs well, obviously you are a more experienced dev than me when it comes to this kind of issue.
About performance there is no significant difference between these two according to Ionic (Capacitor) https://ionic.io/blog/ionic-vs-react-native-performance-comparison

About APIs, well, you have the same access as RN has to those, plugins exist for Capacitor that are native plugins. But about formats and so on, I may not have a good understanding about that.
I was just pointing out that, seeing the maturity of the app, maybe a faster approach for native apps is Capacitor.

Have a nice day!

**Comment by mrjvs** [comment]
Capacitor is great and all if you want to iterate quickly, but we already have a PWA. it gives almost the exact same benefits as a native app with capacitor would give.

The big difference with react native is that react native just kinda gives directions on how it should show. meaning JS only controls what is shown, it doesnt render the components itself.

Capacitor does webview, meaning we still have all web limitations that previously existed. making it no better than running a website. And in terms of performance, it needs to render everything with css/html, not native components. It will still be less tactile than react native.

My main point isn't even performance though, we need to run video:
- JS and parsing video is just a bad shout, it'll need native code. Not possible with ionic to run native code without JS inbetween.
- ionic has CORS restrictions that cannot be turned off. Making it just as good as a browser would be.
- ionic uses webview of the platform, safari has too many limitations to make a good player in a webview (ionic)

**Comment by mrjvs** [comment]
> I can help make the native app

We're planning things internally on how to get it done, this is just the brainstorm issue. No assignee's possible yet.

**Comment by The-Robin-Hood** [comment]
Is @movie-web/providers is supported in react native ? 
Because initially faced a issue - `Property 'atob' doesn't exist`.
Fixed the issue by changing the atob value with encode and decode from base-64 package. But when tried provider.runall with proper params, it always returns null.

**Comment by mrjvs** [comment]
@The-Robin-Hood it's supposed to be compatible, if it's not then we've broken something. Please open an issue and we'll try to fix it

**Comment by LonelyCpp** [comment]
react-native-video would be able to handle steams relatively well, out of the box

**Comment by slytter** [comment]
Would be cool, but I'm almost certain that you will be rejected by Apples app store (possibly Androids too). Their rules are very strict, and apps that finds streams are most likely against their policies (even though content is fetched from third parties).

**Comment by DenuxPlays** [comment]
> @DenuxPlays Yeah, I guess you could do side-loading on Android and soon on iOS in the EU. But I also feel like it's a lot of work just for achieving 'the native feel' to an app that will not be in stores, with the cost being that future maintenance is split between two platforms instead of one. Unless, of course, the project is migrated to React Native with React Native Web. That could work pretty well, I think. RN Web has apparently become more stable.

Yeah but I think there isn't really anything they can do. I really really doubt that google nor apple are gonna approve the app so sie loading or maybe any alternative app stores are the way to go.
Unless I am not right (which would be great tbh) and they can publish it.

**Comment by JorrinKievit** [comment]
We are making it with Expo/RN. And no, no plans to publish to stores (maybe 3rd party ones?)

**Comment by M00NJ** [comment]
It seems you've already decided on how to make it. So just out of curiosity, why was making it available on desktop platforms as well, with something like Flutter not an option?

**Comment by binaryoverload** [comment]
Desktop would not benefit a tremendous lot from a native app. With the upcoming browser extension, that will offer a comparable experience to using a native app in terms of the sources we can offer.

**Comment by castdrian** [comment]
just in case we do ever choose to, flutter would not be necessary, Microsoft maintains a react native extension for macOS and Windows

**Comment by JorrinKievit** [comment]
True, but Linux is also a very small percentage of users

**Comment by The-Robin-Hood** [comment]
@mrjvs If we've started the mobile app work with react native. Is there anyway i can get access to the repo. So i could offer my contribution:)

**Comment by mrjvs** [comment]
@The-Robin-Hood The repo can be found at: https://github.com/movie-web/native-app. Keep in mind however that we we still have the same contribution system where you need to be assigned to an issue before you can make a PR for it.

The issues with the https://github.com/movie-web/movie-web/labels/repo-native-app label are for the native app.

However if you want to seriously help out with the development of the app. I recommend joining our discord server, where we discuss in more detail how we are going to be implemented the app (we have a chat there with all people working on it)

**Comment by DarkCrypt** [comment]
Android TV support would now be possible.

**Comment by vijayvcm** [comment]
Just wanted to check here if the development for native app is still on? Is there any beta version relased for android TV?

**Comment by JorrinKievit** [comment]
> Just wanted to check here if the development for native app is still on? Is there any beta version relased for android TV?

Still being worked on, and we are far from done. We also only have mobile focus for now

**Comment by itsmeadarsh2008** [comment]
You want a Hybrid app. Look 👀 into this [repository](https://github.com/miro-albrecht/pwa-with-native-apis). However, it is unmaintained for over 4 years. It may be helpful for reference.

**Comment by castdrian** [comment]
not necessary, we're using react-native
